debian/patches/broadcom-firmware.patch

```diff
@@ -0,0 +1,152 @@
+Index: b/tools/hciattach.c
+===================================================================
+--- a/tools/hciattach.c	2016-05-26 11:51:11.000000000 -0500
++++ b/tools/hciattach.c	2020-03-08 21:04:47.725900647 -0500
+@@ -62,6 +62,7 @@
+ 	char *bdaddr;
+ 	int  (*init) (int fd, struct uart_t *u, struct termios *ti);
+ 	int  (*post) (int fd, struct uart_t *u, struct termios *ti);
++	char *fwdir;
+ };
+ 
+ #define FLOW_CTL	0x0001
+@@ -275,7 +276,7 @@
+ 
+ static int bcm43xx(int fd, struct uart_t *u, struct termios *ti)
+ {
+-	return bcm43xx_init(fd, u->init_speed, u->speed, ti, u->bdaddr);
++	return bcm43xx_init(fd, u->init_speed, u->speed, ti, u->bdaddr, u->fwdir);
+ }
+ 
+ static int read_check(int fd, void *buf, int count)
+@@ -1226,7 +1227,7 @@
+ {
+ 	printf("hciattach - HCI UART driver initialization utility\n");
+ 	printf("Usage:\n");
+-	printf("\thciattach [-n] [-p] [-b] [-r] [-t timeout] [-s initial_speed]"
++	printf("\thciattach [-n] [-p] [-b] [-r] [-f firmware_directory] [-t timeout] [-s initial_speed]"
+ 			" <tty> <type | id> [speed] [flow|noflow]"
+ 			" [sleep|nosleep] [bdaddr]\n");
+ 	printf("\thciattach -l\n");
+@@ -1244,12 +1245,13 @@
+ 	struct pollfd p;
+ 	sigset_t sigs;
+ 	char dev[PATH_MAX];
++	char *firmware_dir = NULL;
+ 
+ 	detach = 1;
+ 	printpid = 0;
+ 	raw = 0;
+ 
+-	while ((opt=getopt(argc, argv, "bnpt:s:lr")) != EOF) {
++	while ((opt=getopt(argc, argv, "bnpt:s:lrf:")) != EOF) {
+ 		switch(opt) {
+ 		case 'b':
+ 			send_break = 1;
+@@ -1282,6 +1284,10 @@
+ 			raw = 1;
+ 			break;
+ 
++		case 'f':
++			firmware_dir = optarg;
++			break;
++
+ 		default:
+ 			usage();
+ 			exit(1);
+@@ -1358,6 +1364,8 @@
+ 		exit(1);
+ 	}
+ 
++	u->fwdir = firmware_dir;
++
+ 	/* If user specified a initial speed, use that instead of
+ 	   the hardware's default */
+ 	if (init_speed)
+Index: b/tools/hciattach.h
+===================================================================
+--- a/tools/hciattach.h	2015-09-03 20:19:36.000000000 -0500
++++ b/tools/hciattach.h	2020-03-04 20:31:39.000000000 -0600
+@@ -65,4 +65,4 @@
+ int qualcomm_init(int fd, int speed, struct termios *ti, const char *bdaddr);
+ int intel_init(int fd, int init_speed, int *speed, struct termios *ti);
+ int bcm43xx_init(int fd, int def_speed, int speed, struct termios *ti,
+-		const char *bdaddr);
++		const char *bdaddr, const char *fwdir);
+Index: b/tools/hciattach_bcm43xx.c
+===================================================================
+--- a/tools/hciattach_bcm43xx.c	2015-03-11 05:01:57.000000000 -0500
++++ b/tools/hciattach_bcm43xx.c	2020-03-08 20:39:18.603088384 -0500
+@@ -35,6 +35,7 @@
+ #include <dirent.h>
+ #include <time.h>
+ #include <limits.h>
++#include <strings.h>
+ 
+ #include "lib/bluetooth.h"
+ #include "lib/hci.h"
+@@ -42,10 +43,6 @@
+ 
+ #include "hciattach.h"
+ 
+-#ifndef FIRMWARE_DIR
+-#define FIRMWARE_DIR "/etc/firmware"
+-#endif
+-
```

Comment:
I would leave this define in (or rename it to `DEFAULT_FIRMWARE_DIR`) then uses it below.